pulldown-cmark/src/parse.rs

```diff
@@ -831,10 +878,117 @@ impl<'input, F: BrokenLinkCallback<'input>> Parser<'input, F> {
             cur = self.tree[cur_ix].next;
         }
         self.link_stack.clear();
+        self.wikilink_stack.clear();
         self.code_delims.clear();
         self.math_delims.clear();
     }
 
+    /// Handles a wikilink.
+    ///
+    /// This function may bail early in case the link is malformed, so this
+    /// acts as a control flow guard. Returns the link node if a wikilink was
+    /// found and created.
```

Comment:
This seems like a good implementation strategy. I like it! :+1:

It makes it relatively easy to see where the main parser diverges for wikilinks, like having this as its own pass would have, but without potentially generating an improperly nested document (like `<a><a></a></a>`, or, worse, `<a><em></a></em>`).